### PR TITLE
[ICU-739] Fix issue on leaving a team

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix issue on leaving a team where it doesn't succeed. It happens on redirect where channel name (default channel) is on URL while channel's current ID is not set on store.

#### Ticket Link
Jira ticket: [ICU-739](https://mattermost.atlassian.net/browse/ICU-739)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/needs_team/needs_team.jsx [hunk 1]
```diff
@@ -127,6 +127,21 @@ export default class NeedsTeam extends React.Component {
         TeamStore.saveMyTeam(team);
         BrowserStore.setGlobalItem('team', team.id);
         TeamStore.emitChange();
+
+        let channel;
+        if (identifier) {
+            const channelName = identifier.toLowerCase();
+            channel = ChannelStore.getByName(channelName);
```
Comment (enahum) [inline components/needs_team/needs_team.jsx]: Is there a way to use whatever is in redux instead on using the old stores?

File: actions/global_actions.jsx [hunk 2]
```diff
@@ -492,21 +501,23 @@ export async function redirectUserToDefaultTeam() {
...
+        if (channel && channel.team_id === team.id) {
+            selectChannel(channel.id)(dispatch, getState);
+            channelName = channel.name;
         } else if (channelId) {
             const {data} = await getChannelAndMyMember(channelId)(dispatch, getState);
             if (data) {
-                redirect(teams[teamId].name, data.channel.name);
-            } else {
-                redirect(teams[teamId].name, Constants.DEFAULT_CHANNEL);
+                selectChannel(channelId)(dispatch, getState);
```
Comment (saturninoabril) [inline actions/global_actions.jsx]: Setting currentChannelId fix the problem. Others are just code refactor.

File: components/needs_team/needs_team.jsx (hunk 1 above)
Comment (saturninoabril) [reply components/needs_team/needs_team.jsx]: My bad, I thought it's the easy fix. I've updated the PR using the selectId from mattermost-redux.

File: actions/global_actions.jsx [hunk 3]
```diff
@@ -492,21 +501,23 @@ export async function redirectUserToDefaultTeam() {
...
-    if (teams[teamId]) {
+    const team = teams[teamId];
+    if (team) {
         const channelId = BrowserStore.getGlobalItem(teamId);
         const channel = ChannelStore.getChannelById(channelId);
-        if (channel) {
-            redirect(teams[teamId].name, channel.name);
+        let channelName = Constants.DEFAULT_CHANNEL;
+        if (channel && channel.team_id === team.id) {
+            selectChannel(channel.id)(dispatch, getState);
```
Comment (enahum) [inline actions/global_actions.jsx]: Can you do dispatch(selectChannel(channel.id));

File: actions/global_actions.jsx (hunk 2 above)
Comment (enahum) [reply actions/global_actions.jsx]: Same here

File: actions/global_actions.jsx (hunk 3 above)
Comment (saturninoabril) [reply actions/global_actions.jsx]: Oopss, sure. Now updated, thanks!

**Pull request merged**